include/RAJA/policy/simd/forall.hpp

```diff
@@ -63,7 +63,9 @@ RAJA_INLINE void forall_impl(const simd_exec &,
   auto distance = std::distance(begin, end);
   RAJA_SIMD
   for (decltype(distance) i = 0; i < distance; ++i) {
-    loop_body(*(begin + i));
+    using RAJA::internal::thread_privatize;
```

Comment:
I would like someone to explain why we need to privatize at each loop iteration. If there is no performance degradation, then I don't see a problem. Have we compared RAJA::simd_exec policy to a C-style loop with the simd pragma applied? Also, what happens with RAJA::loop_exec policy when a compiler decides to vectorize?

Reply:
Since the simd construct acts like a parallel region that creates SIMD lanes rather than threads, it requires that the user appropriately privatize variables.  For whatever reason there is no firstprivate clause on the simd construct, and we can't use private since the body can't be default constructed, so we're stuck with manually privatizing in each iteration.

Reply:
That's basically what I thought. I think it is still a good idea to see what the performance ramifications are.  What would it mean for the body to be "default constructed"?

Reply:
Given that the privatization step occurs inside the loop, the simd_policy ends up being the slowest policy when compared to OMP+Seq or OMP + loop_exec. For example, I compared the run time of matrix addition and I observed that simd_exec is 4x slower than seq_exec (with 20 threads). Output is correct though.